src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -188,40 +215,55 @@ public Task<Result<PropertyResponse[]>> GetPropertiesAsync(CancellationToken tok
 
         public Task<Result<VersionResponse>> GetVersionAsync(CancellationToken token)
         {
-            const string ServerVersionAPI = "api/server/version"; // Since 2.10; internal
+            // Since 2.10; internal
+            var serverVersionAPI = BuildRelativeUrl("api/server/version");
 
-            return InvokeSonarQubeApi(ServerVersionAPI, token,
+            return InvokeSonarQubeApi(serverVersionAPI, token,
                 stringResponse => new VersionResponse { Version = stringResponse });
         }
 
         public Task<Result<CredentialResponse>> ValidateCredentialsAsync(CancellationToken token)
         {
-            const string ValidateCredentialsAPI = "api/authentication/validate"; // Since 3.3
+            // Since 3.3
+            var validateCredentialsApiUrl = BuildRelativeUrl("api/authentication/validate");
 
-            return InvokeSonarQubeApi(ValidateCredentialsAPI, token,
+            return InvokeSonarQubeApi(validateCredentialsApiUrl, token,
                 stringResponse => new CredentialResponse { IsValid = (bool)JObject.Parse(stringResponse).SelectToken("valid") });
         }
 
-        private static string AppendQueryString(string urlBase, string queryFormat, params object[] args)
+        private Uri BuildRelativeUrl(string relativePath, Dictionary<string, string> queryParameters = null)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>>' instead of 'System.Collections.Generic.Dictionary<string, string>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)